File: backend/app/models/topology.py
from pydantic import BaseModel


class DeviceNode(BaseModel):
    id: str
    label: str
    ip_address: str | None = None
    mac_address: str | None = None
    vendor: str | None = None
    online: bool = True


class TopologyEdge(BaseModel):
    source: str
    target: str


class TopologyGraph(BaseModel):
    nodes: list[DeviceNode]
    edges: list[TopologyEdge]
    source: str  # "stub" for v1.0.0 scaffold, "snmp" once real discovery lands

File: frontend/src/types/topology.ts
export interface DeviceNode {
  id: string
  label: string
  ip_address: string | null
  mac_address: string | null
  vendor: string | null
  online: boolean
}

export interface TopologyEdge {
  source: string
  target: string
  source_port: string | null
  target_port: string | null
}

export interface TopologyGraph {
  nodes: DeviceNode[]
  edges: TopologyEdge[]
  source: string
}

export interface PingResult {
  success: boolean
  latency_ms: number | null
}
